File: pipelines_webinar/pipeline_stack.py
import aws_cdk as cdk
from aws_cdk import (
    aws_codepipeline as codepipeline,
    aws_codepipeline_actions as cpactions,
    pipelines
)
from constructs import Construct
from .webservice_stage import WebServiceStage

APP_ACCOUNT = '391970746680'

class PipelineStack(cdk.Stack):
    def __init__(self, scope: Construct, id: str, **kwargs):
        super().__init__(scope, id, **kwargs)

        source_artifact = codepipeline.Artifact()
        cloud_assembly_artifact = codepipeline.Artifact()

        pipeline = pipelines.CodePipeline(self, 'Pipeline',
            synth=pipelines.ShellStep(
                "Synth",
                input=pipelines.CodePipelineSource.git_hub(
                    "sonuabraham-aremedia/cdk-pipelines-demo",
                    "python",
                    authentication=cdk.SecretValue.secrets_manager("github-oauth-token1"),
                ),
                commands=[
                    "npm install -g aws-cdk",
                    "pip install -r requirements.txt",
                    "pytest unittests",
                    "cdk synth"
                ]
            )
        )

        pre_prod_app = WebServiceStage(self, "Pre-Prod", env={
            "account": APP_ACCOUNT,
            "region": "ap-southeast-2",
        })
        pre_prod_stage = pipeline.add_stage(pre_prod_app)
        pre_prod_stage.add_post(
            pipelines.ShellStep(
                "Integ",
                commands=[
                    "pip install -r requirements.txt",
                    "pytest integtests",
                ],
                env_from_cfn_outputs={
                    "SERVICE_URL": pre_prod_app.url_output
                }
            )
        )

        pipeline.add_stage(WebServiceStage(self, "Prod", env={
            "account": APP_ACCOUNT,
            "region": "ap-southeast-2",
        }))

File: pipelines_webinar/pipeline_stack.ts
import { SecretValue, Stack, StackProps } from "aws-cdk-lib";
import { Construct } from "constructs";
import * as codepipeline from "aws-cdk-lib/aws-codepipeline";
import * as codepipelineActions from "aws-cdk-lib/aws-codepipeline-actions";
import * as pipelines from "aws-cdk-lib/pipelines";
import { WebServiceStage } from "./webservice_stage";
import { CodeBuildStep } from "aws-cdk-lib/pipelines";

export class PipelineStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const sourceArtifact = new codepipeline.Artifact();
    const cloudAssemblyArtifact = new codepipeline.Artifact();

    const sourceAction = new codepipelineActions.GitHubSourceAction({
      actionName: "GitHub",
      output: sourceArtifact,
      oauthToken: SecretValue.secretsManager("github-token"),
      owner: "sonuabraham-aremedia",
      repo: "cdk-pipelines-demo",
      branch: "typescript",
      trigger: codepipelineActions.GitHubTrigger.POLL,
    });

    const synthAction = new CodeBuildStep("Synth", {
      input: pipelines.CodePipelineSource.gitHub(
        "sonuabraham-aremedia/cdk-pipelines-demo",
        "typescript",
        {
          authentication: SecretValue.secretsManager("github-oauth-token1"),
        }
      ),
      installCommands: ["npm install"],
      commands: ["npm run build", "npm test"], // Correctly using commands property
      primaryOutputDirectory: "cdk.out",
    });
    const pipeline = new pipelines.CodePipeline(this, "Pipeline", {
      synth: synthAction,
    });

    // Pre-prod
    const preProdApp = new WebServiceStage(this, "Pre-Prod");
    const preProdStage = pipeline.addStage(preProdApp);
    const serviceUrl = preProdApp.urlOutput;

    preProdStage.addPost(
      new pipelines.ShellStep("IntegrationTests", {
        commands: ["npm install", "npm run build", "npm run integration"],
        envFromCfnOutputs: {
          SERVICE_URL: serviceUrl,
        },
      })
    );

    // Prod
    const prodApp = new WebServiceStage(this, "Prod");
    pipeline.addStage(prodApp);
  }
}
